msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -0,0 +1,185 @@
+		
+		List<SliceSparseMatrix.Triplet> listOfTripletsInSlice = new ArrayList<SliceSparseMatrix.Triplet>(horizontalSlice.values());
+		
+		listOfIntensities = new ArrayList<Double>();
+		
+		for(int i=0;i<listOfTripletsInSlice.size();i++){
+			double intensity = listOfTripletsInSlice.get(i)!=null?listOfTripletsInSlice.get(i).intensity:0;
+			listOfIntensities.add(intensity);
+		}
+				
+		Collections.sort(listOfIntensities);
+		
+		//This is max height for BiGaussian fit. It's in terms of intensities.
+		double maxHeight = listOfIntensities.get(listOfIntensities.size()-1);
+		
+		//Below logic is for finding BiGaussian parameters.		
+		int mu = getScanNumber(maxHeight);
+		int roundedmz = (int) Math.round(mz * 10000); 
+		double halfHeight = (double) maxHeight/2; 
+		
+		BiGaussianParams.maxHeight = maxHeight;
+		BiGaussianParams.mu = mu;
+	
+		double interpolationLeftSideX = InterpolationX(mu,halfHeight,leftBound,rightBound,roundedmz,Direction.LEFT);		
+		//This is sigma left for BiGaussian. 
+		BiGaussianParams.sigmaLeft = (mu - interpolationLeftSideX)/Math.sqrt(2*Math.log(2)); 
+		
+		
+		double interpolationRightSideX = InterpolationX(mu,halfHeight,leftBound,rightBound,roundedmz,Direction.RIGHT);		
+		//This is sigma right for BiGaussian. 
+		BiGaussianParams.sigmaRight = ((interpolationRightSideX - mu))/Math.sqrt(2*Math.log(2));
+		
+	}
+	
+	
+	/**
+	 * <p>
+	 * InterpolationX is used to calculate X value of Halfwidth-halfheight point of either left or right half of BiGaussian.
+	 * </p>
+	 * 
+	 * @param mu a {@link java.lang.Integer} object. This is X value for maximum height in terms of scan number.
+	 * @param halfHeight  a {@link java.lang.Double} object. This is m/z value from the raw file.
+	 * @param leftBound a {@link java.lang.Integer} object. This is minimum scan number.
+	 * @param rightBound a {@link java.lang.Integer} object. This is maximum scan number.
+	 * @param roundedmz a {@link java.lang.Integer} object. This is rounded m/z value.
+	 * @param direction a {@link Enum} object. This decides for which half we want to calculate X value.
+	 */
+	private double InterpolationX(int mu,double halfHeight,int leftBound,int rightBound,int roundedmz,Direction direction){
+		
+		int i = mu;
+		double  interpolationY1 = 0;
+		double interpolationY2 = 0;
+		int previousX2 = direction == Direction.RIGHT ? 1 : -1;
+		
+		while(leftBound <= i && i <= rightBound) {
+			
+			i += previousX2;
+			SliceSparseMatrix.Triplet triplet1 = horizontalSlice.get(i,roundedmz);
+			
+			if(triplet1!=null){
+				interpolationY1 = triplet1.intensity;
+			}
+			
+			if(interpolationY1 <halfHeight && triplet1!=null){
+				SliceSparseMatrix.Triplet triplet2 =  horizontalSlice.get(i-previousX2,roundedmz);
+				if(triplet2!=null){
+					interpolationY2 = triplet2.intensity;
+					break;
+				}
+			}
+			
+		}
+		
+		double interpolationX =  ((halfHeight - interpolationY1)*((i-previousX2)-i)
+				/(interpolationY2 - interpolationY1))+i;
```

Comment:
Instead of an exception, it's better to test whether ```interpolationY1 == interpolationY1```. Also, isn't the final ```+i``` outside of the division?